src/nodes/process.py

```diff
@@ -0,0 +1,392 @@
+    def __init__(
+        self,
+        streams: StdStreams,
+    ):
+        super().__init__(streams=streams)
+        self.fee_recipient: ChecksumAddress | None = None
+
+    async def get_process(self) -> LighthouseVCProcess:
+        # Wait a bit to ensure that the execution and consensus nodes are started
+        startup_interval = 10
+        await asyncio.sleep(startup_interval)
+
+        # Wait for nodes to be ready
+        await wait_for_execution_node()
+        await wait_for_consensus_node()
+
+        chain_state = await get_chain_finalized_head()
+        await wait_execution_catch_up_consensus(chain_state)
+
+        # Fetch mev escrow address and cache it
+        if not self.fee_recipient:
+            self.fee_recipient = await self._get_fee_recipient()
+
+        lighthouse_dir = settings.nodes_dir / 'lighthouse'
+
+        validator_definitions_path = (
+            settings.nodes_dir / 'lighthouse' / 'validators' / 'validator_definitions.yml'
+        )
+        # Create the parent directory if it does not exist
+        if not validator_definitions_path.parent.exists():
+            validator_definitions_path.parent.mkdir(parents=True, exist_ok=True)
+
+        # Usually the validator definitions file is created during `import` command
+        # `lighthouse account validator import ...`
+        # The problem is the case of per-keystore password files.
+        # Natively, Lighthouse import does not support per-keystore password files.
+        # So we need to update the validator definitions file manually.
+
+        logger.info('Updating validator definitions file %s...', validator_definitions_path)
+        update_validator_definitions_file(
+            keystore_files=LocalKeystore.list_keystore_files(),
+            output_path=validator_definitions_path,
+        )
+        # Note on slashing protection.
+        # Normally, slashing protection database is updated during `import` command
+        # `lighthouse account validator import ...`
+        # But since we update the validator definitions file manually, we need to ensure
+        # that slashing protection database is updated as well.
+        # The option `init_slashing_protection` helps to achieve that.
+        # Otherwise, validator client will refuse to start.
+        init_slashing_protection = True
+
+        return LighthouseVCProcess(
+            network=settings.network,
+            lighthouse_dir=lighthouse_dir,
+            fee_recipient=self.fee_recipient,
+            streams=self.streams,
+            init_slashing_protection=init_slashing_protection,
+        )
+
+    async def _get_fee_recipient(self) -> ChecksumAddress:
+        """
+        Fetches the fee recipient address from the vault contract.
+        This is used to set the suggested fee recipient for the validator client.
+        """
+        vault_contract = VaultContract(settings.vault)
+        fee_recipient = await vault_contract.mev_escrow()
```

Comment:
Seems like this call will fail in case we specify zero address as vault address initially